tests/base/src/test/java/org/keycloak/tests/admin/client/AbstractClientSearchTest.java

```diff
@@ -0,0 +1,100 @@
+package org.keycloak.tests.admin.client;
+
+import org.keycloak.models.ClientProvider;
+import org.keycloak.representations.idm.ClientRepresentation;
+import org.keycloak.testframework.annotations.InjectClient;
+import org.keycloak.testframework.annotations.InjectRealm;
+import org.keycloak.testframework.realm.ClientConfig;
+import org.keycloak.testframework.realm.ClientConfigBuilder;
+import org.keycloak.testframework.realm.ManagedClient;
+import org.keycloak.testframework.realm.ManagedRealm;
+import org.keycloak.testframework.runonserver.InjectRunOnServer;
+import org.keycloak.testframework.runonserver.RunOnServerClient;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+
+public class AbstractClientSearchTest {
+
+    @InjectRealm
+    ManagedRealm testRealm;
+
+    static final String CLIENT1_REF = "client1";
+    static final String CLIENT2_REF = "client2";
+    static final String CLIENT3_REF = "client3";
+
+    @InjectClient(config = ClientSearchTest.Client1Config.class, ref = CLIENT1_REF)
+    ManagedClient client1;
+
+    @InjectClient(config = ClientSearchTest.Client2Config.class, ref = CLIENT2_REF)
+    ManagedClient client2;
+
+    @InjectClient(config = ClientSearchTest.Client3Config.class, ref = CLIENT3_REF)
+    ManagedClient client3;
```

Comment:
Strictly speaking the tests doesn't need the managed client, so could just have done a single realm config that adds the 3 clients instead of separate calls to add/remove